CMakeLists.txt

```diff
@@ -109,6 +114,66 @@ set (PIO_FILESYSTEM_HINTS IGNORE CACHE STRING "Filesystem hints (lustre or gpfs)
 option (PIO_ENABLE_TESTS  "Enable the testing builds"                           ON)
 option (PIO_VALGRIND_CHECK  "Enable memory leak check using valgrind"           OFF)
 
+#==============================================================================
+#  Z5 BACKEND
+#==============================================================================
+
+# Z5
+if (DEFINED Z5_DIR)
+  message("Z5_DIR: " ${Z5_DIR})
+  set(Z5_DIR CACHE PATH "z5 location")
+  include_directories(${Z5_DIR})
+  #boost
+  find_package(Boost REQUIRED COMPONENTS system filesystem)
+  message("Boost_FILESYSTEM_LIBRARY:" ${Boost_FILESYSTEM_LIBRARY})
+  message("Boost_SYSTEM_LIBRARY:" ${Boost_SYSTEM_LIBRARY})
+  set(BOOST_LIBRARIES
+          ${Boost_FILESYSTEM_LIBRARY}
+          ${Boost_SYSTEM_LIBRARY})
+  #zlib
+  option(WITH_ZLIB ON)
+  if(WITH_ZLIB)
+    find_package(ZLIB REQUIRED)
+    set(ZLIB_ROOT CACHE PATH "zlib location")
+    include_directories(${ZLIB_INCLUDE_DIRS})
+    message("ZLIB_INCLUDE_DIRS: " ${ZLIB_INCLUDE_DIRS})
+    add_definitions(-DWITH_ZLIB)
+    SET(COMPRESSION_LIBRARIES "${COMPRESSION_LIBRARIES};${ZLIB_LIBRARIES}")
+  endif()
+  option(WITH_BLOSC ON)
+  if(WITH_BLOSC)
+    find_package(BLOSC REQUIRED)
+    set(BLOSC_ROOT CACHE PATH "blosc location")
+    include_directories(${BLOSC_INCLUDE_DIR})
+    message("BLOSC_INCLUDE_DIRS: " ${BLOSC_INCLUDE_DIR})
+    message("BLOSC_LIBRARY: " ${BLOSC_LIBRARIES})
+    add_definitions(-DWITH_BLOSC)
+    SET(COMPRESSION_LIBRARIES "${COMPRESSION_LIBRARIES};${BLOSC_LIBRARIES}")
+  endif()
+  #lz4
+  option(WITH_LZ4 ON)
+  if(WITH_LZ4)
+    find_package(LZ4 REQUIRED)
+    set(LZ4_ROOT CACHE PATH "lz4 location")
+    include_directories(${LZ4_INCLUDE_DIR})
+    add_definitions(-DWITH_LZ4)
+    SET(COMPRESSION_LIBRARIES "${COMPRESSION_LIBRARIES};${LZ4_LIBRARY}")
+  endif()
+  #CXX 11
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++14")
```

Comment:
This seems like a compiler specific flag, which compiler(s) have you tested with?

Reply:
This is for gnu/8.3.0. I tested on intel as well. The flag is 
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -cxxlib=/glade/u/home/weile/miniconda3/envs/z5-py36/ -pthread /glade/u/home/weile/miniconda3/envs/z5-py36/lib/libstdc++.so")